plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -159,8 +164,15 @@ func (s *SQLServer) gatherServer(server string, query Query, acc telegraf.Accumu
 	}
 	defer conn.Close()
 
+	ctx := context.Background()
```

Comment:
Put this as the first line of the Gather function, then pass the single context down into this function.  This way we only need to setup (and cancel) the timeout once for all servers.